### PR TITLE
Kafka: SASL/OAuthbearer Implementation

### LINKED ISSUE #3681: Apache Kafka Scaler: SASL/OAuthbearer Implementation
### Proposal

Add support for Kafka SASL/OAuthbearer authentication.
https://kafka.apache.org/documentation/#security_sasl_oauthbearer

### Use-Case

We often deploy our consumers/producers outside of the Kafka infrastructure. Because of this we are looking for ways to use OAuth Providers like Azure AD or KeyCloak with the Keda Apache Kafka Scalers.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by DevinGillman**
Added support for SASL/OAuthbearer authentication.

https://github.com/kedacore/keda-docs/pull/942

Fixes #3681

### CODE REVIEW

**Review by zroubalik — COMMENTED** [review]
Looking good!

**Comment by zroubalik** [comment]
@DevinGillman could you please remove the unrelated commit from this PR?

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 ### Improvements
 
 - **General:** Add explicit seccompProfile type to securityContext config ([#3561](https://github.com/kedacore/keda/issues/3561))
+- **Apache Kafka Scaller:** SASL/OAuthbearer Implementation ([#3681](https://github.com/kedacore/keda/issues/3681))
```
[the remark this answers is not in the document]
Comment (DevinGillman) [reply CHANGELOG.md]: Fixed

**Comment by DevinGillman** [comment]
Unrelated commit removed.

**Pull request closed without merging**